lib/modules/manager/poetry/extract.ts

```diff
@@ -68,35 +83,39 @@ function extractFromSection(
       lockedVersion = poetryLockfile[packageName];
     }
     if (!is.string(currentValue)) {
-      const version = currentValue.version;
-      const path = currentValue.path;
-      const git = currentValue.git;
-      if (version) {
-        currentValue = version;
-        nestedVersion = true;
-        if (!!path || git) {
-          skipReason = path ? 'path-dependency' : 'git-dependency';
-        }
-      } else if (path) {
+      if (is.array(currentValue)) {
```

Comment:
what's the reason to change the order of this code?

Reply:
We need to do this check first, otherwise we won't be able to access the properties of `currentValue` as it can be an array.

![image](https://github.com/renovatebot/renovate/assets/12210067/a352e465-bad4-40ae-8695-70009a1943b2)